src/matter/common/MessageExchange.ts

```diff
@@ -201,6 +225,9 @@ export class MessageExchange<ContextT> {
             ackPromise = promise;
```

Comment:
Suggestion: let's have ackPromise return the message containing the message so we can crash if expectAckOnly is set and this message is not a standalone ack.